apps/api/src/workspace/service/workspace.service.ts

```diff
@@ -542,4 +513,42 @@ export class WorkspaceService {
       })) > 0
     )
   }
+
+  /**
+   * Creates a search query object based on provided project IDs and search term.
+   * @param model The Prisma model name (e.g., 'Project', 'Environment')
+   * @param projectIds The IDs of projects to query
+   * @param searchTerm The search term to query by
+   * @param fields The fields to apply the search term in the query
+   * @returns The prisma query object
+   * @private
+   */
+  private createSearchQuery<T>(
```

Comment:
I think this got a bit over-generalized. We just want this function to return the `where` and `select` clause since these fields are similar. We don't want to pass in the model name aswell. Ideally, we would like to perform the prisma table operation in each of the individual function itself, but just use the query generator to generate `where` and `select` into a single object.

Reply:
I tried to exactly do this in the second last commit [327abe0](https://github.com/keyshade-xyz/keyshade/pull/455/commits/327abe0c8a9a39f03c4507862fb183baa8c4501b) , but that was causing issues because the `id` field in the `where` clause is a bit different for the `Project` model and the rest of them.

For the `Project` Query
```typescript
where : {
  id : { in : projectIds },
  // ...rest
}
```
For `Screct`, `Environment` and `Variable` Queries
```typescript
where : {
  project : { id : { in : projectIds } },
  // ...rest
}
```
That's why I added the `model` param for checking this. Do you thing is there a way to do this without the model param?

Reply:
Hmmm, thats an overlook from my end. Okay drop it then, please revert back before this change. Sorry about the inconvenience.

Reply:
No worries at all :) 
I have reverted the changes.